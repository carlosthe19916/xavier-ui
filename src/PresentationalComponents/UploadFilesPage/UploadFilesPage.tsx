import React, { Fragment, Component } from 'react';
import { withRouter } from 'react-router-dom';

import {
    Main,
    PageHeader,
    PageHeaderTitle
} from '@redhat-cloud-services/frontend-components';
import {
    Breadcrumb,
    BreadcrumbItem
} from '@patternfly/react-core';
import { GlobalProps } from '../../models/GlobalProps';

interface Props extends GlobalProps {
    title: string;
    mainStyle?: any;
    showBreadcrumb?: boolean;
};

interface State {
}

export class UploadFilesPage extends Component<Props, State> {

    toIndex(event: any): void {
        event.preventDefault();
        this.props.history.push('/upload');
    }

    showRootLink() {
        return (
            this.props.history && this.props.history.location.pathname !== '/upload' ?
                <BreadcrumbItem to='#' onClick={ this.toIndex }>Reports</BreadcrumbItem>
                : ''
        );
    }

    showBreadcrumb() {
        return (
            this.props.showBreadcrumb && <Breadcrumb style={ { marginLeft: 'calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)' } }>
                { this.showRootLink() }
                <BreadcrumbItem isActive style={ { marginTop: 0 } }>{ this.props.title }</BreadcrumbItem>
            </Breadcrumb>
        );
    }

    render() {
        const { title, children } = this.props;

        return (
            <Fragment>
                <PageHeader>
                    { this.showBreadcrumb() }
                    <PageHeaderTitle title={ title } />
                </PageHeader>
                <Main style={ this.props.mainStyle }>
                    { children }
                </Main>
            </Fragment>
        );
    }
};

export default withRouter(UploadFilesPage);
