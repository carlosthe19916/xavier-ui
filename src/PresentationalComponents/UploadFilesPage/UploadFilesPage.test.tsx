import { shallow } from "enzyme";
import { UploadFilesPage } from "./UploadFilesPage";

describe("UploadFilesPage", () => {
  const routerProps = {
    history: null,
    location: null,
    match: null
  };

  it("expect to render without breadcrumb", () => {
    const wrapper = shallow(
      <UploadFilesPage title="My title" {...routerProps}>
        children
      </UploadFilesPage>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render with breadcrumb and without showing rootLink", () => {
    const props = Object.assign({}, routerProps, {
      history: {
        location: {
          pathname: "/upload"
        }
      }
    });

    const wrapper = shallow(
      <UploadFilesPage title="My title" showBreadcrumb={true} {...props}>
        children
      </UploadFilesPage>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render with breadcrumb and showing rootLink", () => {
    const props = Object.assign({}, routerProps, {
      history: {
        location: {
          pathname: "/mypath"
        }
      }
    });

    const wrapper = shallow(
      <UploadFilesPage title="My title" showBreadcrumb={true} {...props}>
        children
      </UploadFilesPage>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
