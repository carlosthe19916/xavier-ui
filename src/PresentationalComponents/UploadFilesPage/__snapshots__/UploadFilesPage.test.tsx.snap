// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFilesPage expect to render with breadcrumb and showing rootLink 1`] = `
<Fragment>
  <PageHeader>
    <Breadcrumb
      aria-label="Breadcrumb"
      className=""
      style={
        Object {
          "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
        }
      }
    >
      <BreadcrumbItem
        className=""
        component="a"
        isActive={false}
        onClick={[Function]}
        target={null}
        to="#"
      >
        Reports
      </BreadcrumbItem>
      <BreadcrumbItem
        className=""
        component="a"
        isActive={true}
        style={
          Object {
            "marginTop": 0,
          }
        }
        target={null}
        to={null}
      >
        My title
      </BreadcrumbItem>
    </Breadcrumb>
    <PageHeaderTitle
      title="My title"
    />
  </PageHeader>
  <Connect(Main)>
    children
  </Connect(Main)>
</Fragment>
`;

exports[`UploadFilesPage expect to render with breadcrumb and without showing rootLink 1`] = `
<Fragment>
  <PageHeader>
    <Breadcrumb
      aria-label="Breadcrumb"
      className=""
      style={
        Object {
          "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
        }
      }
    >
      <BreadcrumbItem
        className=""
        component="a"
        isActive={true}
        style={
          Object {
            "marginTop": 0,
          }
        }
        target={null}
        to={null}
      >
        My title
      </BreadcrumbItem>
    </Breadcrumb>
    <PageHeaderTitle
      title="My title"
    />
  </PageHeader>
  <Connect(Main)>
    children
  </Connect(Main)>
</Fragment>
`;

exports[`UploadFilesPage expect to render without breadcrumb 1`] = `
<Fragment>
  <PageHeader>
    <PageHeaderTitle
      title="My title"
    />
  </PageHeader>
  <Connect(Main)>
    children
  </Connect(Main)>
</Fragment>
`;
